packages/components/src/Filters/FilterInput.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from 'react';
+import { Checkbox, Radio } from '@patternfly/react-core';
+
+export interface FilterInputProps {
+  addRemoveFilters?: (value?: string, param?: string, type?: string, checked?: boolean) => void;
+  className?: string;
+  currentPage?: string;
+  id: string;
+  label: React.ReactNode;
+  param: string;
+  value?: string;
+  filters?: Record<string, string>;
+  type?: string;
+  onChange?: (checked: boolean, event: React.FormEvent<HTMLInputElement>) => void;
+}
+
+const FilterInput: React.FC<FilterInputProps> = ({ addRemoveFilters, param, filters, type = 'checkbox', value, id, label }) => {
+  const [checked, setChecked] = useState(false);
+  useEffect(() => {
+    if (type !== 'radio') {
+      if (filters && value && param in filters && filters[param] && filters[param].includes(value)) {
+        setChecked(true);
+      } else {
+        setChecked(false);
+      }
+    }
+  });
+  const handleChange = () => {
+    switch (type) {
+      case 'checkbox':
+        setChecked(!checked);
+        break;
+    }
+    addRemoveFilters && addRemoveFilters(value, param, type, checked);
```

Comment:
Let's use safe accessor here as well.
```suggestion
    addRemoveFilters?.(value, param, type, checked);
```

Reply:
@aneelac22 any opinion on this one? I am fine with keeping the shortcutting, it just feels a bit more natural to use safe accessor.